From: Stefanos Harhalakis <v13@v13.gr>
Subject: Fix gcc warnings.
 This is just to make -Werror succeed as the samples are not that important.

Index: libnet-1.1.6/sample/cdp.c
===================================================================
--- libnet-1.1.6.orig/sample/cdp.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/cdp.c	2012-06-21 20:55:20.992874861 +0100
@@ -86,7 +86,7 @@
     index += 5;
 
     /* this TLV is handled by the libnet builder */
-    value = argv[2];
+    value = (u_char *)argv[2];
     len = strlen(argv[2]);
 
     /* build CDP header */
Index: libnet-1.1.6/sample/dhcp_discover.c
===================================================================
--- libnet-1.1.6.orig/sample/dhcp_discover.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/dhcp_discover.c	2012-06-21 20:56:41.951290580 +0100
@@ -238,7 +238,8 @@
         fprintf(stderr, "Packets sent:  %lld\n"
                     "Packet errors: %lld\n"
                     "Bytes written: %lld\n",
-                    ls.packets_sent, ls.packet_errors, ls.bytes_written);
+                    (long long)ls.packets_sent, (long long)ls.packet_errors,
+		    (long long)ls.bytes_written);
         libnet_destroy(l);
         
         free(options);
Index: libnet-1.1.6/sample/ping_of_death.c
===================================================================
--- libnet-1.1.6.orig/sample/ping_of_death.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/ping_of_death.c	2012-06-21 20:58:05.241747888 +0100
@@ -158,7 +158,8 @@
     fprintf(stderr, "Packets sent:  %lld\n"
                     "Packet errors: %lld\n"
                     "Bytes written: %lld\n",
-                    ls.packets_sent, ls.packet_errors, ls.bytes_written);
+                    (long long)ls.packets_sent, (long long)ls.packet_errors,
+		    (long long)ls.bytes_written);
     libnet_destroy(l);
     free(data);
     return (EXIT_SUCCESS);
Index: libnet-1.1.6/sample/udp1.c
===================================================================
--- libnet-1.1.6.orig/sample/udp1.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/udp1.c	2012-06-21 20:58:31.749211851 +0100
@@ -204,7 +204,8 @@
     fprintf(stderr, "Packets sent:  %lld\n"
                     "Packet errors: %lld\n"
                     "Bytes written: %lld\n",
-                    ls.packets_sent, ls.packet_errors, ls.bytes_written);
+                    (long long)ls.packets_sent, (long long)ls.packet_errors,
+		    (long long)ls.bytes_written);
     libnet_destroy(l);
     return (EXIT_SUCCESS);
 bad:
Index: libnet-1.1.6/sample/udp2.c
===================================================================
--- libnet-1.1.6.orig/sample/udp2.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/udp2.c	2012-06-21 21:01:24.792242599 +0100
@@ -137,7 +137,7 @@
                 cport,                              /* destination port */
                 LIBNET_UDP_H + payload_s,           /* packet size */
                 0,                                  /* checksum */
-                payload,                            /* payload */
+                (uint8_t *)payload,                 /* payload */
                 payload_s,                          /* payload size */
                 l,                                  /* libnet handle */
                 udp);                               /* libnet id */
@@ -208,13 +208,15 @@
 #endif
 
     libnet_timersub(&e, &s, &r);
-    fprintf(stderr, "Total time spent in loop: %d.%d\n", r.tv_sec, r.tv_usec);
+    fprintf(stderr, "Total time spent in loop: %d.%d\n", (int)r.tv_sec,
+		    (int)r.tv_usec);
 
     libnet_stats(l, &ls);
     fprintf(stderr, "Packets sent:  %lld\n"
                     "Packet errors: %lld\n"
                     "Bytes written: %lld\n",
-                    ls.packets_sent, ls.packet_errors, ls.bytes_written);
+                    (long long)ls.packets_sent, (long long)ls.packet_errors,
+		    (long long)ls.bytes_written);
     libnet_destroy(l);
     return (EXIT_SUCCESS);
 bad:
Index: libnet-1.1.6/sample/icmp6_echoreq.c
===================================================================
--- libnet-1.1.6.orig/sample/icmp6_echoreq.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/icmp6_echoreq.c	2012-06-21 21:02:07.579992513 +0100
@@ -126,7 +126,7 @@
 			char payload[56];
 			int i;
 			for (i=0; i<56; i++) payload[i]='A'+((char)(i%26));
-			t = libnet_build_icmpv4_echo(ICMP6_ECHO,0,0,1,0,payload,sizeof(payload),l,t);
+			t = libnet_build_icmpv4_echo(ICMP6_ECHO,0,0,1,0,(uint8_t *)payload,sizeof(payload),l,t);
 
             if (build_ip)
             {
Index: libnet-1.1.6/sample/icmp_echo_cq.c
===================================================================
--- libnet-1.1.6.orig/sample/icmp_echo_cq.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/icmp_echo_cq.c	2012-06-21 21:02:39.515552361 +0100
@@ -149,7 +149,7 @@
             0,                                    /* checksum */
             src_ip,                               /* source IP */
             dst_ip,                               /* destination IP */
-            payload,                              /* payload */
+            (uint8_t *)payload,                   /* payload */
             payload_s,                            /* payload size */
             l,                                    /* libnet handle */
             0);
Index: libnet-1.1.6/sample/dot1x.c
===================================================================
--- libnet-1.1.6.orig/sample/dot1x.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/dot1x.c	2012-06-21 21:03:18.708239579 +0100
@@ -68,7 +68,7 @@
             0,
             LIBNET_802_1X_PACKET,
             sizeof(payload),
-            payload,
+            (uint8_t *)payload,
             sizeof(payload),
             l,
             0);
Index: libnet-1.1.6/sample/dns.c
===================================================================
--- libnet-1.1.6.orig/sample/dns.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/dns.c	2012-06-21 21:03:37.982577603 +0100
@@ -140,7 +140,7 @@
 	0,             /* num_anws_rr */
 	0,             /* num_auth_rr */
 	0,             /* num_addi_rr */
-	payload,
+	(uint8_t *)payload,
 	payload_s,
 	l,
 	0
Index: libnet-1.1.6/sample/rpc_tcp.c
===================================================================
--- libnet-1.1.6.orig/sample/rpc_tcp.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/rpc_tcp.c	2012-06-21 21:04:00.556973551 +0100
@@ -131,7 +131,7 @@
         0,
         0,
         NULL,
-        payload,
+        (uint8_t *)payload,
         payload_s,
         l,
         0);
Index: libnet-1.1.6/sample/rpc_udp.c
===================================================================
--- libnet-1.1.6.orig/sample/rpc_udp.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/rpc_udp.c	2012-06-21 21:04:24.933401169 +0100
@@ -136,7 +136,7 @@
         0,
         0,
         NULL,
-        payload,
+        (uint8_t *)payload,
         payload_s,
         l,
         0);
Index: libnet-1.1.6/sample/mpls.c
===================================================================
--- libnet-1.1.6.orig/sample/mpls.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/mpls.c	2012-06-21 21:07:43.425885468 +0100
@@ -107,8 +107,8 @@
                 }
                 break;
             case 'p':
-                payload = optarg;
-                payload_s = strlen(payload);
+                payload = (u_char *)optarg;
+                payload_s = strlen((char *)payload);
                 break;
             default:
                 exit(EXIT_FAILURE);
Index: libnet-1.1.6/sample/fddi_tcp1.c
===================================================================
--- libnet-1.1.6.orig/sample/fddi_tcp1.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/fddi_tcp1.c	2012-06-21 21:08:20.399534941 +0100
@@ -132,7 +132,7 @@
         0,                                          /* checksum */
         0,                                          /* urgent pointer */
         LIBNET_TCP_H + payload_s,                   /* TCP packet size */
-        payload,                                    /* payload */
+        (uint8_t *)payload,                         /* payload */
         payload_s,                                  /* payload size */
         l,                                          /* libnet handle */
         0);                                         /* libnet id */
Index: libnet-1.1.6/sample/fddi_tcp2.c
===================================================================
--- libnet-1.1.6.orig/sample/fddi_tcp2.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/fddi_tcp2.c	2012-06-21 21:08:42.129916715 +0100
@@ -132,7 +132,7 @@
         0,                                          /* checksum */
         0,                                          /* urgent pointer */
         LIBNET_TCP_H + payload_s,                   /* TCP packet size */
-        payload,                                    /* payload */
+        (uint8_t *)payload,                         /* payload */
         payload_s,                                  /* payload size */
         l,                                          /* libnet handle */
         0);                                         /* libnet id */
Index: libnet-1.1.6/sample/tring_tcp1.c
===================================================================
--- libnet-1.1.6.orig/sample/tring_tcp1.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/tring_tcp1.c	2012-06-21 21:09:13.066460312 +0100
@@ -132,7 +132,7 @@
         0,                                          /* checksum */
         0,                                          /* urgent pointer */
         LIBNET_TCP_H + payload_s,                   /* TCP packet size */
-        payload,                                    /* payload */
+        (uint8_t *)payload,                         /* payload */
         payload_s,                                  /* payload size */
         l,                                          /* libnet handle */
         0);                                         /* libnet id */
Index: libnet-1.1.6/sample/tring_tcp2.c
===================================================================
--- libnet-1.1.6.orig/sample/tring_tcp2.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/tring_tcp2.c	2012-06-21 21:09:38.263903139 +0100
@@ -132,7 +132,7 @@
         0,                                          /* checksum */
         0,                                          /* urgent pointer */
         LIBNET_TCP_H + payload_s,                   /* TCP packet size */
-        payload,                                    /* payload */
+        (uint8_t *)payload,                         /* payload */
         payload_s,                                  /* payload size */
         l,                                          /* libnet handle */
         0);                                         /* libnet id */
Index: libnet-1.1.6/sample/bgp4_hdr.c
===================================================================
--- libnet-1.1.6.orig/sample/bgp4_hdr.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/bgp4_hdr.c	2012-06-21 21:10:24.918723213 +0100
@@ -116,8 +116,8 @@
 		break;
 
 	    case 'p':
-		payload = optarg;
-		payload_s = strlen(optarg);
+		payload = (u_char *)optarg;
+		payload_s = strlen((char *)optarg);
 		break;
 
             default:
Index: libnet-1.1.6/sample/bgp4_open.c
===================================================================
--- libnet-1.1.6.orig/sample/bgp4_open.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/bgp4_open.c	2012-06-21 21:11:01.794371558 +0100
@@ -124,7 +124,7 @@
 		break;
 
 	    case 'p':
-		payload = optarg;
+		payload = (u_char *)optarg;
 		break;
 
 	    case 'S':
@@ -156,7 +156,7 @@
 
     if (payload && !payload_s)
     {
-	payload_s = strlen(payload);
+	payload_s = strlen((char *)payload);
     }
 
     length = LIBNET_BGP4_OPEN_H + payload_s;
Index: libnet-1.1.6/sample/bgp4_update.c
===================================================================
--- libnet-1.1.6.orig/sample/bgp4_update.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/bgp4_update.c	2012-06-21 21:13:40.853169540 +0100
@@ -95,7 +95,7 @@
                                                                                \
     if (ptr && !size)                                                          \
     {                                                                          \
-	size = strlen(ptr);                                                    \
+	size = strlen((char *)ptr);                                            \
     }                                                                          
 
 
@@ -169,12 +169,12 @@
                 break;
 
 	    case 'p':
-		payload = optarg;
-		payload_s = strlen(payload);
+		payload = (u_char *)optarg;
+		payload_s = strlen((char *)payload);
 		break;
 
 	    case 'w':
-		withdraw_rt = optarg;
+		withdraw_rt = (u_char *)optarg;
 		break;
 
 	    case 'W':
@@ -182,7 +182,7 @@
 		break;
 
 	    case 'a':
-		attr = optarg;
+		attr = (u_char *)optarg;
 		break;
 
 	    case 'A':
@@ -190,7 +190,7 @@
 		break;
 
 	    case 'i':
-		info = optarg;
+		info = (u_char *)optarg;
 		break;
 
 	    case 'I':
Index: libnet-1.1.6/sample/bgp4_notification.c
===================================================================
--- libnet-1.1.6.orig/sample/bgp4_notification.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/bgp4_notification.c	2012-06-21 21:14:15.948787222 +0100
@@ -123,8 +123,8 @@
 		break;
 
 	    case 'p':
-		payload = optarg;
-		payload_s = strlen(optarg);
+		payload = (u_char *)optarg;
+		payload_s = strlen((char *)optarg);
 		break;
 
             default:
Index: libnet-1.1.6/sample/gre.c
===================================================================
--- libnet-1.1.6.orig/sample/gre.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/gre.c	2012-06-21 21:16:05.998724822 +0100
@@ -177,7 +177,7 @@
 		gre_flags|=GRE_CSUM;
                 break;
 	    case 'r':
-		routing = optarg;
+		routing = (u_char *)optarg;
 		gre_flags|=GRE_ROUTING;
                 break;
 	    case 'k':
@@ -316,11 +316,11 @@
 	    fprintf(stderr, "Can't build GRE last SRE header: %s\n", libnet_geterror(l));
 	    goto bad;
 	}
-	size += LIBNET_GRE_SRE_H + strlen(routing);
+	size += LIBNET_GRE_SRE_H + strlen((char *)routing);
 	t = libnet_build_gre_sre(
 	    GRE_IP,                                 /* address family */
 	    0,                                      /* offset */
-	    strlen(routing),                        /* routing length */
+	    strlen((char *)routing),                /* routing length */
 	    routing,                                /* routing info */
 	    NULL,                                   /* payload */
 	    0,                                      /* payload size */
@@ -379,8 +379,8 @@
     } 
 
     t = libnet_autobuild_ethernet(
-            "11:11:11:11:11:11",                                    /* ethernet destination */
-            ETHERTYPE_IP,                          /* protocol type */
+            (uint8_t *)"11:11:11:11:11:11",         /* ethernet destination */
+            ETHERTYPE_IP,                           /* protocol type */
             l);                                     /* libnet handle */
     if (t == -1)
     {
Index: libnet-1.1.6/sample/ip_link.c
===================================================================
--- libnet-1.1.6.orig/sample/ip_link.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/ip_link.c	2012-06-21 21:17:15.126942483 +0100
@@ -88,8 +88,8 @@
 		break;
 
 	    case 'p':
-		strncpy(payload, optarg, sizeof(payload)-1);
-		payload_s = strlen(payload);
+		strncpy((char *)payload, optarg, sizeof(payload)-1);
+		payload_s = strlen((char *)payload);
 		break;
 
 	    case 'h':
@@ -154,8 +154,8 @@
     }
 
     eth_ptag = libnet_build_ethernet(
-        hwdst,                                      /* ethernet destination */
-        hwsrc,                                      /* ethernet source */
+        (uint8_t *)hwdst,                           /* ethernet destination */
+        (uint8_t *)hwsrc,                           /* ethernet source */
         ETHERTYPE_IP,                               /* protocol type */
         NULL,                                       /* payload */
         0,                                          /* payload size */
Index: libnet-1.1.6/sample/ip_raw.c
===================================================================
--- libnet-1.1.6.orig/sample/ip_raw.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/ip_raw.c	2012-06-21 21:17:48.082523128 +0100
@@ -81,8 +81,8 @@
 		break;
 
 	    case 'p':
-		strncpy(payload, optarg, sizeof(payload)-1);
-		payload_s = strlen(payload);
+		strncpy((char *)payload, optarg, sizeof(payload)-1);
+		payload_s = strlen((char *)payload);
 		break;
 
 	    case 'h':
Index: libnet-1.1.6/sample/sebek.c
===================================================================
--- libnet-1.1.6.orig/sample/sebek.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/sebek.c	2012-06-21 21:21:55.309881960 +0100
@@ -142,8 +142,8 @@
 
 	    case 'p':
 		payload_flag = 1;
-		payload = optarg; 
-		payload_s = strlen(payload);
+		payload = (u_char *)optarg; 
+		payload_s = strlen((char *)payload);
 		break;
 
 	    case 'h':
@@ -194,7 +194,7 @@
 
     if (!payload)
     {
-	payload = cmd;
+	payload = (uint8_t *)cmd;
 	payload_s = length;
     }
 
@@ -209,9 +209,9 @@
 	pid,
 	uid,
 	fd,
-	cmd,
+	(uint8_t *)cmd,
 	/* LIBNET_ETH_H + LIBNET_IPV4_H + LIBNET_UDP_H + LIBNET_SEBEK_H +*/ length,
-	payload,
+	(uint8_t *)payload,
 	payload_s,
 	l,
 	0
@@ -261,9 +261,9 @@
     }
 
     
-    eth_dst = libnet_hex_aton(eth_dst, &c);
+    eth_dst = (char *)libnet_hex_aton((char *)eth_dst, &c);
     ptag = libnet_autobuild_ethernet(
-	eth_dst,                                /* ethernet destination */
+	(uint8_t *)eth_dst,                     /* ethernet destination */
 	ETHERTYPE_IP,                           /* protocol type */
 	l);                                     /* libnet handle */
 
Index: libnet-1.1.6/sample/hsrp.c
===================================================================
--- libnet-1.1.6.orig/sample/hsrp.c	2012-03-19 16:59:50.000000000 +0000
+++ libnet-1.1.6/sample/hsrp.c	2012-06-21 21:22:50.157849682 +0100
@@ -94,7 +94,7 @@
     }
 
     memset(authdata, 0, 8);
-    strncpy(authdata, "cisco", 5);
+    strncpy((char *)authdata, "cisco", 5);
 
 
     ptag = libnet_build_hsrp(
@@ -158,9 +158,9 @@
     }
 
     
-    eth_dst = libnet_hex_aton(eth_dst, &c);
+    eth_dst = (char *)libnet_hex_aton(eth_dst, &c);
     ptag = libnet_autobuild_ethernet(
-	eth_dst,                                /* ethernet destination */
+	(uint8_t *)eth_dst,                     /* ethernet destination */
 	ETHERTYPE_IP,                           /* protocol type */
 	l);                                     /* libnet handle */
 
